### PR TITLE
tests: ames+gall subscribe/kick/resubscribe sequence

### PR DESCRIPTION
**Pull request by belisarius222**
Adds a test that goes through the sequence of subscribing to a remote ship, getting kicked, sending a cork, and resubscribing.  Checks that the first Ames bone was correctly corked and that Gall incremented the subscription nonce.

### CODE REVIEW

File: pkg/arvo/sys/vane/ames.hoon [hunk 1]
```diff
@@ -2747,6 +2747,7 @@
       ::
       ++  on-sink-cork
         ^+  peer-core
+        ~>  %slog.0^leaf+"ames: got cork {<[her.channel bone]>}"
```
Comment (belisarius222) [inline pkg/arvo/sys/vane/ames.hoon:2750]: I should revert this

**Review by Fang- — APPROVED** [review]
Nice, very good to get these! This looks fine to me. Also nice to have the verbosity toggle sitting right there. Might be a way to refactor the `*-check-*` calls a bit, but the clarity of literal declarations of in/outputs is probably better.

File: pkg/arvo/tests/sys/grq.hoon [hunk 2]
```diff
@@ -0,0 +1,492 @@
...
+      :-  ~[//unix]
+      :*  %hear  [%& ~bud]
+          0x5f.f966.8e00.0449.bdec.9006.c7e5.1237.
+          1d87.53fe.d7bb.ad00.0100.0223.c6a8.5800
+      ==
+      :~  :-  :~  /sys/way/~bud/pub
+                  /use/sub/0w1.d6Isf/out/~bud/pub/1/sub-foo/~bud
+                  /init
+              ==
+          [%give %done ~]
```
Comment (belisarius222) [inline pkg/arvo/tests/sys/grq.hoon:459]: I need to write a handler for the `%done`, too.

**Review by Fang- — APPROVED** [review]
Good to have these!

**Pull request merged**